purchase_minimum_amount/models/purchase_order.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    minimum_po_amount = fields.Float(
+        related='partner_id.minimum_po_amount',
+        string="Purchase Minimum Amount",
+        readonly=True,
+    )
+
+    @api.model
+    def _check_minimum_amount(self):
+        po_amt_block = self.env.ref('purchase_minimum_amount.'
+                                    'minimum_amount_block_reason',
+                                    raise_if_not_found=False)
+        under_min = self.amount_untaxed < self.minimum_po_amount
```

Comment:
There is an inconsistency between decorator (model) and self usage.

Either you decorate with api.multi and iterate or ensure_one(), or you don't decorate (even if you come from the create() which is model decorated, 'po' is a recordset)